### DISCUSSION TITLE
Is it possible to subscribe to windows messages loop in win ui 3?

### DISCUSSION THREAD

**Discussion by roshannzth** [microsoft/microsoft-ui-xaml#8934] [Q&A] [answered]
I am trying to implement minimizing window to tray, and I see some challenges for myself.
One old win32 solution is subscribing to the windows loop for SC_MINIMIZE event and using shell notify to execute the task.

So i looked on the web and i could not find much info,  how do I subscribe to windows loop in win ui.
This can help other use cases like mousemove or keyboards etc

**Comment by castorix** [comment, marked as answer]
> how do I subscribe to windows loop in win ui.

With [SetWindowSubclass](https://learn.microsoft.com/en-us/windows/win32/controls/subclassing-overview#setwindowsubclass) (from _WinRT.Interop.WindowNative.GetWindowHandle_ in C#)
(I used it for example in [WinUI3_NotifyIcon](https://github.com/castorix/WinUI3_NotifyIcon)  mainly to handle the custom Tray message (_WM_TRAYMOUSEMESSAGE_ in [MainWindow.xaml.cs](https://github.com/castorix/WinUI3_NotifyIcon/blob/master/MainWindow.xaml.cs))

**Reply by roshannzth** [reply]
So this does exactly what i asked for,
Here is an sample c++ code that i wrote, hopefully it will help others too


        MainWindow thisWindow = *this;
        winrt::com_ptr<IWindowNative> windowNative = thisWindow.as<IWindowNative>();

        //Get the HWND for the XAML Window
        static HWND hWnd;
        windowNative->get_WindowHandle(&hWnd);

        SetWindowSubclass(hWnd, &mySubClassProc, 1, 0);

And callback method can be as follows
```
    LRESULT CALLBACK MainWindow::mySubClassProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam, UINT_PTR uIdSubclass, DWORD_PTR dwRefData)
    {
        switch (uMsg)
        {
        case WM_MOUSEMOVE:
            break;

        case WM_SYSCOMMAND:
            if ((wParam & 0xfff0) == SC_MINIMIZE)
            {
                // The window is being minimized
                // Handle the SW_MINIMIZE event here
            }
            break;

        case WM_LBUTTONDOWN:
            break;

        case WM_NCDESTROY:
            RemoveWindowSubclass(hWnd, &mySubClassProc, uIdSubclass);
            break;
        }

        return DefSubclassProc(hWnd, uMsg, wParam, lParam);
    }
```
